Internationalization i18n (Multi Language Support)

v1lling opened the pull request:
References Issue #271

This pull request implements internalisation using the i18next library.
All labels are moved to the provided i18n file `en.json'.
This deprecates customLabel and page-specific labels like page.title / page.description and field.label.
For backwards compatibility, if these labels are defined in config.js, they will override the i18n files.

dsternlicht commented:
@v1lling overall looking good, but I don't like the deprecation of many properties that allows you to set placeholders and texts. I believe that the right approach would be that hard coded properties will rewrite i18n values.

WDYT?

v1lling commented:
@dsternlicht 
Thank you for your feedback! I just want to make sure I understand your concern correctly.
Additionally to `customLabels` this PR also deprecates:

- page.title and page.description
- field.label

The reasoning behind this change is to allow translations for page.title and page.description, which appear at the top of each page, as well as for field labels. In my use case, translating these texts is essential.

With this approach, the i18n JSON files can define these texts per page and field while maintaining backward compatibility. If, for example, a field has a label defined in config.json, that text will take precedence, ensuring no breaking changes.

> I believe that the right approach would be that hard coded properties will rewrite i18n values.

Could you elaborate on this?

dsternlicht commented:
@v1lling Thanks for the info. I understand the intention of the feature, but I'm not sure about the deprecation of widely used features.

So how about a different approach - use the i18n files if exist - else, use the `page.title`, `page.description`, etc.

In the documentation you can recommend using i18n files rather than setting up the inline props instead of deprecating them.

WDYT?

v1lling commented:
@dsternlicht Thanks for the suggestion!
This is actually more or less how it currently works. If page.title, page.description, or field.label are defined in the config.js, those values take precedence. Only if these fields are not defined, the values from the i18n files are used.

I originally marked them as deprecated in the README because I thought it would be better to define these texts in the i18n files from the start. However, I have now removed the deprecation warning and changed it to a recommendation instead. Does this make sense to you?

dsternlicht commented:
@v1lling Yes, definitely! 🎉
